### DISCUSSION TITLE
Does stb_image know what format the image is?

### DISCUSSION THREAD

**Discussion by Raildex** [nothings/stb#1344] [Q&A] [answered]
I have a memory buffer, that contains arbitrary images - I don't know in advance whether it's 8 bit, 16 bit or float.
Can stb_image distinguish it automatically or do I need to test whether the decoding is succesful for each format?

**Comment by nothings** [comment, marked as answer]
`stbi_is_16_bit_from_memory`
`stbi_is_hdr_from_memory`
